demo/admin/src/userGroups/userGroupAdditionalItemFields.ts

```diff
@@ -1,4 +1,7 @@
-const userGroupAdditionalItemFields = {
+import { BlocksBlockAdditionalItemField, ListBlockAdditionalItemField } from "@comet/blocks-admin";
+import { GQLUserGroup } from "@src/graphql.generated";
+
+const userGroupAdditionalItemFields: { userGroup: ListBlockAdditionalItemField<GQLUserGroup> | BlocksBlockAdditionalItemField<GQLUserGroup> } = {
```

Comment:
If `additionalItemFields` in either `createListBlock` or `createBlocksBlock` is typed correctly, the fields will be inferred correctly.

Reply:
wouldn't it be possible to infer this type?

Reply:
I mean to omit the type here

Reply:
It is possible. However, `userGroup` is then inferred as string, and not as `GQLUserGroup`:

```
Type 'BlocksBlockItem<BlockInterface<any, any, any, any>, { userGroup: string; }>' is not assignable to type 'UserGroupItem'.
  Types of property 'userGroup' are incompatible.
    Type 'string' is not assignable to type 'GQLUserGroup'.ts(2322)
```

Reply:
you could possibly use `"All" as GQLUserGroup`?

Reply:
Do you think an as cast is better than the explicit typing?

Reply:
Generally of course not.

But here, if you convert a generic string to an strings-union (don't know how this type is called) the type gets more accurate and can be inferred.

And inferring is in this case better than explicitly defining imho. Especially this `ListBlockAdditionalItemField<GQLUserGroup> | BlocksBlockAdditionalItemField<GQLUserGroup>` is not very nice.

Reply:
Now I understand! Fixed by b5dc8af4bd337de1694d171229e4422f730030f8

> ListBlockAdditionalItemField<GQLUserGroup> | BlocksBlockAdditionalItemField<GQLUserGroup>

Should I remove those two types from the public API as well?

Reply:
> Should I remove those two types from the public API as well?

if we don't need it anymore, better remove it